File: 2021/08/1.pl
#!/usr/bin/perl

my $ns=0;
while (<>)
{
    @input=split /\|/;

    @res=split(" ",@input[1]);

    print "i1: @input[1] res: @res\n";

    
    foreach my $out (@res)
    {
	my $len=length($out);
	
	if ($len == 2) 
	{
	    $ns++;
	} 
	if ($len == 4)
	{
	    $ns++;
	}
	if ($len == 3)
	{
	    $ns++;
	}
	if ($len == 7)
	{
	    $ns++;
	}
	print "l: $out $len $ns\n";
    }

}

File: 2022/21/1.py
#!/usr/bin/python3
import re
from sympy import solve

monkeys={}
f=open('input')
for l in f.read().strip().split("\n"):
    monkey,value=l.split(": ")
    monkeys[monkey]=value
        
def findeq(monkey):
    if monkey=="humn":
        return " h "
    if re.search("\d+",monkeys[monkey]):
        return monkeys[monkey]
    else:
        l,op,r=monkeys[monkey].split(" ")
        return "(" + findeq(l) + op + findeq(r) + ")"

h=int(monkeys["humn"])
print("p1",int(eval(findeq("root"))))
monkeys["root"]= monkeys["root"].replace("+","-")
print("p2:",solve(findeq("root"))[0])
